applications/minotari_node/src/stratum/block_template_repository.rs

```diff
@@ -0,0 +1,329 @@
+
+        let mut total_excess = UncompressedCommitment::default();
+        let mut total_nonce = UncompressedPublicKey::default();
+        let range_proof_type = RangeProofType::RevealedValue;
+        let (_, coinbase_output, coinbase_kernel, wallet_output) = generate_coinbase_with_wallet_output(
+            0.into(),
+            MicroMinotari::from(reward),
+            height,
+            &CoinBaseExtra::try_from(vec![])?,
+            &key_manager,
+            &script_key_id,
+            &solo_address,
+            true,
+            self.consensus_rules.consensus_constants(height),
+            range_proof_type,
+            MemoField::new_open(vec![], TxType::Coinbase).expect("empty user-data should always be valid"),
+        )
+        .await?;
+
+        new_template.body.add_output(coinbase_output);
+        let new_nonce = key_manager
+            .get_next_key(TransactionKeyManagerBranch::KernelNonce.get_branch_key())
+            .await?;
+
+        total_nonce = &total_nonce +
+            &new_nonce
+                .pub_key
+                .to_public_key()
+                .map_err(|e| anyhow!("Failed to get public key: {}", e))?;
+        total_excess = &total_excess +
+            &coinbase_kernel
+                .excess
+                .to_commitment()
+                .map_err(|e| anyhow!("Failed to get commitment: {}", e))?;
+        let (spending_key_id, nonce) = (wallet_output.commitment_mask_key_id, new_nonce.key_id);
+        let kernel_message = TransactionKernel::build_kernel_signature_message(
+            &TransactionKernelVersion::get_current_version(),
+            coinbase_kernel.fee,
+            coinbase_kernel.lock_height,
+            &coinbase_kernel.features,
+            &None,
+        );
+        let mut kernel_signature = UncompressedSignature::default();
+        kernel_signature = &kernel_signature +
+            &key_manager
+                .get_partial_txo_kernel_signature(
+                    &spending_key_id,
+                    &nonce,
+                    &CompressedPublicKey::new_from_pk(total_nonce.clone()),
+                    &CompressedPublicKey::new_from_pk(total_excess.as_public_key().clone()),
+                    &TransactionKernelVersion::get_current_version(),
+                    &kernel_message,
+                    &coinbase_kernel.features,
+                    TxoStage::Output,
+                )
+                .await?
+                .to_schnorr_signature()
+                .map_err(|e| anyhow!("Failed to get schnorr signature: {}", e))?;
+        let kernel_new = KernelBuilder::new()
+            .with_fee(0.into())
+            .with_features(coinbase_kernel.features)
+            .with_lock_height(coinbase_kernel.lock_height)
+            .with_excess(&CompressedCommitment::from_commitment(total_excess))
+            .with_signature(CompressedSignature::new_from_schnorr(kernel_signature))
+            .build()
+            .unwrap();
```

Comment:
`unwrap` should be handled
```suggestion
            .unwrap();
```